lua/treesitter-context/context.lua

```diff
@@ -182,6 +169,29 @@ end
 
 local M = {}
 
+---@param bufnr integer
+---@param row integer
+---@param col integer
+---@return LanguageTree[]
+local function get_parent_langtrees(bufnr, range)
```

Comment:
Does this work properly for nested injections? it looks to me this will only return at most 2 langtrees. The root and the most inner.

Reply:
You are right, I misinterpreted `language_for_range` implementation. I changed the code to manually traverse children langtrees, please take a look.

To test the nested injection, I came up with the markdown test case replacing HTML one. Context both for `<html>` and `<script>` displays properly when I run vim manually, but in tests injecting language in Markdown seems to be broken (look at `test` job for the reference). Do you have any idea how we could fix it?

Reply:
@lewis6991 It took me a while, but I finally get it to work.

Looks like tests are not performing full initialization of `nvim-treesitter` plugin, but just `require'nvim-treesitter.configs'.setup { ... }`. Because of that some modules are not initialized, including  `nvim-treesitter.query_predicates` which contains `set-lang-from-info-string!` directive configuration required by Markdown queries.

My straightforward solution for that is to explicitly initialize the plugin via `require'nvim-treesitter'.setup()`. I don't think it is the greatest solution, as we end up invoking `setup()` twice (just on different modules), which seems weird, but I couldn't find any other way that would not induce `setup` duplication.